packages/core/content-manager/admin/src/features/DocumentContext.tsx

```diff
@@ -55,12 +61,33 @@ const DocumentContextProvider = ({
    */
   const [currentDocumentMeta, changeDocument] = React.useState<DocumentMeta>(initialDocument);
   const document = useDocument(currentDocumentMeta);
+  const [backButtonHistory, setBackButtonHistory] = React.useState<DocumentMeta[]>([]);
+
+  // Handlers to add, remove and reset back button history
+  const addDocumentToHistory = (document: DocumentMeta) => {
+    setBackButtonHistory((prev) => [...prev, document]);
+  };
+  const removeDocumentFromHistory = () => {
+    const lastDocument = backButtonHistory[backButtonHistory.length - 1];
+    setBackButtonHistory((prev) => prev.slice(0, prev.length - 1));
+    return lastDocument;
+  };
+  const resetHistory = () => {
+    setBackButtonHistory([]);
+  };
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
 
   return (
     <DocumentProvider
       changeDocument={changeDocument}
       document={document}
       meta={currentDocumentMeta}
+      backButtonHistory={backButtonHistory}
+      addDocumentToHistory={addDocumentToHistory}
+      removeDocumentFromHistory={removeDocumentFromHistory}
+      resetHistory={resetHistory}
+      isModalOpen={isModalOpen}
+      setIsModalOpen={setIsModalOpen}
```

Comment:
We shouldn't just overload the context with everything. Can't the relation modal handle most of the history stuff? I really don't want the modal state in this provider.